crates/node/src/networking/p2p/pea2pea.rs

```diff
@@ -163,16 +167,19 @@ impl P2P {
         source: SocketAddr,
         req: protocol::DiagnosticsRequestKind,
     ) -> io::Result<()> {
-        let resp = protocol::DiagnosticsResponseV0::Version {
-            major: env!("CARGO_PKG_VERSION_MAJOR").parse::<u16>().unwrap(),
-            minor: env!("CARGO_PKG_VERSION_MINOR").parse::<u16>().unwrap(),
-            patch: env!("CARGO_PKG_VERSION_PATCH").parse::<u16>().unwrap(),
-            build: format!(
-                "{}: {}",
-                env!("VERGEN_BUILD_TIMESTAMP"),
-                env!("VERGEN_GIT_DESCRIBE")
-            ),
-        };
+        let resp = protocol::Message::V0(protocol::MessageV0::DiagnosticsResponse(
+            self.public_node_key,
+            protocol::DiagnosticsResponseV0::Version {
+                major: env!("CARGO_PKG_VERSION_MAJOR").parse::<u16>().unwrap(),
```

Comment:
These variable are only set is you use cargo. Node are not always started with cargo run?
The other possibility is to use the last commit that can be set in the code during the build process.

Reply:
These are build into the binary:
- https://github.com/gevulotnetwork/gevulot/blob/main/crates/node/build.rs#L2-L5
- https://github.com/gevulotnetwork/gevulot/blob/main/Containerfile#L6-L7